File: Core-backend-layer/Programming-fundamentals/Python/DesignPatterns/src/factoryMethod/grocery_store.py
from factoryMethod.store import Store
from factoryMethod.cashier import Cashier

class GroceryStore(Store):
    def __init__(self, store_name):
        super().__init__(store_name)

    def create_worker(self, worker_name):
        # The factory method for a grocery store creates cashiers
        return Cashier(worker_name)

File: Core-backend-layer/Programming-fundamentals/Java/DesignPatterns/src/factoryMethod/GroceryStore.java
package factoryMethod;

public class GroceryStore extends Store {
    public GroceryStore(String storeName) {
        super(storeName);
    }

    @Override
    protected Worker createWorker(String workerName) {
        // The factory method for a grocery store creates cashiers
        return new Cashier(workerName);
    }
}